### DISCUSSION TITLE
fail to generate an X.509 certificate

### DISCUSSION THREAD

**Discussion by chiachin2686** [open-quantum-safe/openssl#305] [Q&A] [answered]
I followed the steps to build and run in OQS-OpenSSL_1_1_1-stable. But, when I used TLS demo to generate an X.509 certificate, I got an error message.  

using the following command:
`apps/openssl req -x509 -new -newkey oqs_kem_default -keyout oqs_kem_default_CA.key -out oqs_kem_default_CA.crt -nodes -subj "/CN=oqstest CA" -days 365 -config apps/openssl.cnf`

Got an error:
`Unknown algorithm oqs_kem_default`

I also followed the guide (oqsdefault & Code Generation), but still got an error message.      
How can I fix it?  Thank you for your help!!

**Comment by baentsch** [comment, marked as answer]
You can only use a [signature algorithm](https://github.com/open-quantum-safe/openssl#authentication) to create a certificate, not a KEM algorithm. Thus, try "oqs_sig_default" instead.

**Reply by chiachin2686** [reply]
I appreciate your help very much.  I use "oqs_sig_default" is OK.  But, there is another problem. When I use a signature algorithm without an asterisk, I got the same error message.

Got an error:
`Unknown algorithm picnicl1fs`

I followed the "Code Generation" guide step by step to change the enable value from false to true. But, still got the same error. 

There is my steps:
1. Change the enable value of that variant in "oqs-template/generate.yml"  from false to true
2. Run `python3 oqs-template/generate.py`
3. Run `make generate_crypto_objects` to re-generate the object-related files

Did I miss some steps?

**Reply by baentsch** [reply]
No, this looks OK -- maybe it'd be a good idea to also run `make clean; make` to be sure all is re-built. After that, you may want to run `openssl speed test` to get an output of all truly enabled and available algorithms.

**Reply by chiachin2686** [reply]
I run `make clean; make` and then the signature algorithm can use.  I am thankful for your assistance.
